blog/2023-06-20-Deployment-vs-Release.md

```diff
@@ -0,0 +1,56 @@
+
+![He's not Deployment, I am!](./images/2023-06-20-Deployment-vs-Release/not-fred.gif)
+
+Deployment, an operational activity, involves installing and manipulating the configuration of an environment. Ideally this is an internal activity with frequent occurrence. Release, however, is a strategic process concerning the planning and timing of delivery to users, impacting end-users directly, and can happen (less) frequently following the cadence of the organisation. The Deployment takes us to a state we wish to be in, and the release makes it available for use by end-users.
+
+To break it down, deployment is the putting of the thing you made (artefact, container, microservice, infrastructure, serverless function etc), in the place you want it to go e.g., An environment built to run and support interactions with your deployable thing. This isn't always as easy as it sounds, and each deployable will have it's own challenges and so will the environments you are working with.
+
+The purpose of a deployment is to validate performance under real-world conditions, deliver value by rolling out new features and updates, streamline and automate updates to minimise disruptions, and facilitate rollbacks to maintain system stability. This can be done in multiple ways, from manual (in...dev...right) to fully automated (hurray!). It could involve a few steps, or it could be a complex pipeline involving multiple stages, jobs, teams, and even geographical zones. Whether the most simple, or the most complex, it's always worth remembering that every deployment needs good testing, version control, and rollback capabilities.  
+
+OK so that's deployments, what about releases? A release is where functionality e.g., a new feature or an upgraded version of an API, is made available to its users. This is all around _when_ the new functionality is presented to the users. The timing and execution of a release is the most important consideration, commonly balancing the needs of the business e.g., getting new feature out that generates revenue, and the expectations of the users e.g., uptime.  It serves to evolve the product, meet user expectations, measure user response for future development refinement, and mitigate risk by controlling the timing and manner of delivery.
+
+The common expectation is a successful deployment is followed immediately by a release, and while this can happen, you might want to deploy a new version without releasing it to your users. There are many ways this can be done, remembering this can be dependant on what you are deploying and the underlying infrastructure of your environment(s):
```

Comment:
```suggestion
### Deployment types
The common expectation is a successful deployment is followed immediately by a release, and while this can happen, you might want to deploy a new version without releasing it to your users. There are many ways this can be done, remembering this can be dependant on what you are deploying and the underlying infrastructure of your environment(s):
```